Assets/FullInspector2/Modules/StoragesManager/fiStoragesManager.cs

```diff
@@ -0,0 +1,199 @@
+                }
+            }
+
+            //latodo this should be on the PR as well (the DontDestroyOnLoad check)
+            if (!scene.IsValid() || !scene.isLoaded || scene.name == "DontDestroyOnLoad") {
+                return PrefabStorage;
+            }
+
+            return _Storages[scene];
+        }
+
+
+        /// <summary>
+        /// This will ensure that <see cref="_Storages"/> always has fresh
+        /// references for each storage in the loaded scenes.
+        /// </summary>
+        /// <param name="forced"></param>
+        private void RefreshStorages(bool forced = false) {
+            //always refresh at least once
+            if (_didInitialRefresh && !UpdateLastSeenScenes()) {
+                // ReSharper disable once ConditionIsAlwaysTrueOrFalse
+                var hasNullStorages = _Storages.Any(s => s.Value == null || s.Value.gameObject == null);
+                var mustRefresh = false;
+                if (!hasNullStorages) {
+                    if (_Storages.Count == 0) {
+                        mustRefresh = true;
+                    } else {
+                        for (var i = 0; i < SceneManager.sceneCount; i++) {
+                            var scene = SceneManager.GetSceneAt(i);
+                            if (scene.isLoaded) {
+                                if (_Storages.ContainsKey(scene)) {
+                                    continue;
+                                }
+
+                                //we'll have to refresh if we stored keys for scenes that were unloaded
+                            } else if (!_Storages.ContainsKey(scene)) {
+                                continue;
+                            }
+
+                            mustRefresh = true;
+                            break;
+                        }
+                    }
+                }
+
+                if (!forced && !mustRefresh && !hasNullStorages) {
+                    return;
+                }
+            }
+
+            _didInitialRefresh = true;
+
+            //get fresh references for all storages that already exist in the currently loaded scenes
+            _Storages.Clear();
+            foreach (var found in Object.FindObjectsOfType<T>()) {
+                var scene = found.gameObject.scene;
+                if (!_Storages.Keys.Contains(scene)) {
+                    _Storages[scene] = found;
+                } else {
+                    //Debug.LogWarning("scene " + scene.name + " already has a storage, but found another. Destroying...");
+                    Object.DestroyImmediate(found.gameObject);
+                    if (!fiLateBindings.EditorApplication.isPlaying) {
+                        fiLateBindings.SceneManagement.EditorSceneManager.MarkSceneDirty(scene);
+                    }
+                }
+            }
+
+            for (var i = 0; i < SceneManager.sceneCount; i++) {
+                var scene = SceneManager.GetSceneAt(i);
```

Comment:
FYI: var should only be used when the type is obvious (ie, already in the line) - otherwise it decreases readability.